classic/frame/runtime/dlp_runtime.c

```diff
@@ -212,3 +213,17 @@ dlp_thread_set_num_threads(md_t n_threads)
     // env variables will not be of effect going forward.
     dlp_tl_rntm.ext_mt_ctr_var = FALSE;
 }
+
+void
+dlp_version_query(int* major, int* minor, int* patch)
+{
+    char* AOCL_DLP_VERSION_MAJOR_STR = AOCL_DLP_VERSION_MAJOR;
+    char* AOCL_DLP_VERSION_MINOR_STR = AOCL_DLP_VERSION_MINOR;
+    char* AOCL_DLP_VERSION_PATCH_STR = AOCL_DLP_VERSION_PATCH;
+    if (major)
+        *major = atoi(AOCL_DLP_VERSION_MAJOR_STR);
+    if (minor)
+        *minor = atoi(AOCL_DLP_VERSION_MINOR_STR);
+    if (patch)
+        *patch = atoi(AOCL_DLP_VERSION_PATCH_STR);
+}
```

Comment:
These intermediate string variables are unnecessary.

```suggestion
void
dlp_version_query(int* major, int* minor, int* patch)
{
    if (major)
        *major = atoi(AOCL_DLP_VERSION_MAJOR);
    if (minor)
        *minor = atoi(AOCL_DLP_VERSION_MINOR);
    if (patch)
        *patch = atoi(AOCL_DLP_VERSION_PATCH);
}
```

Reply:
Changes made.